includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1226 @@
+			'amount'  => number_format( $amount, 2, '.', '' ),
+			'pending' => false,
+		);
+
+		return $data;
+	}
+
+	/**
+	 * Builds an array of line items/totals to be sent back to Square in the lineItems array.
+	 *
+	 * @since x.x.x
+	 * @param array $totals
+	 * @return array
+	 */
+	public function build_payment_request_line_items( $totals = array() ) {
+		// Ignoring nonce verification checks as it is already handled in the parent function.
+		$totals     = empty( $totals ) ? $this->get_cart_totals() : $totals;
+		$line_items = array();
+		$order_id   = isset( $_POST['order_id'] ) ? (int) sanitize_text_field( wp_unslash( $_POST['order_id'] ) ) : absint( get_query_var( 'order-pay' ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+
+		if ( $order_id ) {
+			$order    = wc_get_order( $order_id );
+			$iterable = $order->get_items();
+		} else {
+			$iterable = WC()->cart->get_cart();
+		}
+
+		foreach ( $iterable as $item ) {
+			$amount = number_format( $order_id ? $order->get_subtotal() : $item['line_subtotal'], 2, '.', '' );
+
+			if ( $order_id ) {
+				$quantity_label = 1 < $item->get_quantity() ? ' x ' . $item->get_quantity() : '';
+			} else {
+				$quantity_label = 1 < $item['quantity'] ? ' x ' . $item['quantity'] : '';
+			}
+
+			$item = array(
+				'label'   => $order_id ? $item->get_name() . $quantity_label : $item['data']->get_name() . $quantity_label,
+				'amount'  => $amount,
+				'pending' => false,
+			);
+
+			$line_items[] = $item;
+		}
+
+		if ( $totals['shipping'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Shipping', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['shipping'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['taxes'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Tax', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['taxes'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['discount'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Discount', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['discount'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['fees'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Fees', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['fees'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		return $line_items;
+	}
+
+	/**
+	 * Get the payment request object in an ajax request
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function ajax_get_payment_request() {
+		check_ajax_referer( 'wc-cash-app-get-payment-request', 'security' );
+
+		$payment_request = array();
+
+		try {
+			$payment_request = $this->get_payment_request();
+
+			if ( empty( $payment_request ) ) {
+				/* translators: Context (product, cart, checkout or page) */
+				throw new \Exception( esc_html__( 'Empty payment request data for page.', 'woocommerce-square' ) );
```

Comment:
Is this added to a log file or shown directly to a shopper?

Reply:
This is shown in the browser console (log) if debug mode on the checkout page is enabled and not directly to the shopper. No error will be logged if the debug mode is OFF.